src/ast/nodes/ObjectExpression.ts

```diff
@@ -169,8 +188,18 @@ export default class ObjectExpression extends NodeBase {
 			const property = this.properties[index];
 			if (kinds.indexOf(property.kind) < 0) continue;
 			if (property.computed) {
-				properties.push(property);
-			} else if ((<Identifier>property.key).name === name) {
+				const value = property.key.getLiteralValueAtPath([]);
+				if (String(value) === name) {
```

Comment:
But value is not guaranteed to be a string. However, it will be cast to string when treated as a property key. Just as accessing `x[1]` and `x["1"]` is equivalent, `{[1]: true}`  and `{["1"]: true}` have the same shape. And `x["[object Object]"]` is equivalent to `x[{}]` 😜.

I have also added a test for this behaviour.

Reply:
`===` won't case `value` into a string if it isn't already, so I think we'd be safe.

My worry was exactly the converse - what if name is `[object Object]`!

Reply:
(I must admit I did also test this out earlier...)